### ISSUE TITLE
Spacemacs can not start correctly after upgrading to 0.105.0

### ISSUE THREAD

**Issue by tommyjiang** [syl20bnr/spacemacs#4375]
The error is as following:

    error: Package `spacemacs-theme-' is unavailable.

I just deleted the whole directory `.emacs.d` and re-install Spacemacs but it still exists.

It feels like my config `.spacemacs.d` is conflict with the new version. I am trying to figure out what is the problem.

**Comment by StreakyCobra** [comment]
CC @syl20bnr as it was already discussed in #4097. Also I saw this error once [on gitter](https://gitter.im/syl20bnr/spacemacs?at=566017089a969fd24f3c4a8a) but without solution.

**Comment by StreakyCobra** [comment]
Can you also try running `emacs --insecure` and see if it helps?

**Comment by tommyjiang** [comment]
Thanks for your advice @StreakyCobra. The problem still exists even I delete the `elpa` directory. Maybe I should consider using the older version..

**Comment by StreakyCobra** [comment]
Let's wait for @syl20bnr then :-)

**Comment by tommyjiang** [comment]
Thanks for your kind help @StreakyCobra

**Comment by syl20bnr** [comment]
@tommyjiang did you try launching emacs with `emacs --insecure` ? Should fix the issue (I have the exact same at work).

**Comment by syl20bnr** [comment]
We should document it clearly by mentioning it on README.md and in the FAQ/

**Comment by tommyjiang** [comment]
@syl20bnr I use proxy to connect `melpa.org` and it seems all right with the backup version `0.104.8` from another Mac. I will try to test whether it works fine with the newest version, i.e., `0.105.0`.

**Comment by syl20bnr** [comment]
@tommyjiang in `0.104.8` ELPA packages are fetched with `http`. It is now `https` by default unless you either:
- launch emacs with `emacs --insecure`
- set the variable `dotspacemacs-elpa-https` to `nil`

**Comment by tommyjiang** [comment]
I tested the newest version `0.105.0` with `proxychains-ng` on Mac. Everything seems OK.

So I suggetst the connection problem without proxy is the critical reason for this. For me maybe the website is blocked by GFW of China. For others, maybe it is related to @syl20bnr's explanation with `https` in the newest version.

Hope my solution helps.

**Comment by BohrShaw** [comment]
It's probably not related to the network or specially `https`, as Spacemacs starts fine on v0.104.8 while stucks on v0.105.

**Comment by et2010** [comment]
I had the same error while try to fix issue #4402 , I forgot to remove the `--insecure` argument after I removed my `.emacs.d` directory and start installing spacemacs from the very beginning. That's when the error popped up. After I removed the argument, everthing is fine except #4402 still haunts me.

**Comment by royseto** [comment]
I am seeing this error too. Running with emacs --insecure does not help.

Environment:
* Fresh build of emacs 24.5.1 from source
* Ubuntu 14.04 running in a Docker container
* Running at home in California (no proxy). I confirmed using "curl https://melpa.org" that I can access that URL over https without problems.

Steps I am taking:
1. cd $HOME
2. rm -rf .emacs .spacemacs .emacs.d
3. git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
4. run either "emacs" or "emacs --insecure"

My emacs Warnings buffer has the following contents:
```
Warning (initialization): An error occurred while loading `/home/dev/.emacs.d/init.el':

error: Package `spacemacs-theme-' is unavailable

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the `--debug-init' option to view a complete error backtrace.
```

Spacemacs 0.104.8 had been working fine for me (on emacs 24.3.1).

**Comment by syl20bnr** [comment]
Wow github, I did not want to close it so soon, shouldn't have put the fixes tag in comment :-)

**Comment by syl20bnr** [comment]
I consider this fixed as it is now mentioned in the documentations but we can let it opened since there may be several issues.

**Comment by royseto** [comment]
I suspect this may not be completely fixed. I started Emacs with the `--insecure` argument as mentioned in commit 74689e0 but am still seeing the behavior I described in my earlier comment in this issue. This reproduces on 0.105.1 as well as 0.105.0. When I revert to 0.104.8, spacemacs loading works fine again.

This blocks spacemacs from starting for me on 0.105.1 and 0.105.0.

**Comment by BohrShaw** [comment]
`--insecure` doesn't help too.

**Comment by StreakyCobra** [comment]
I suppose the problem on your side is emacs **24.3**. We know there are bugs with emacs version older than 24.4, and this is probably one of them. We are dropping support of it as no regular contributor is working on it.

**Comment by BohrShaw** [comment]
@StreakyCobra I'm on Emacs 24.5 on Windows 10.

**Comment by StreakyCobra** [comment]
Sorry, I was speaking for @royseto

**Comment by HIKALU-Z** [comment]
yes . we meet the same problem~

**Comment by royseto** [comment]
@StreakyCobra 

Yes, I had a number of issues blocking Spacemacs from working after upgrading from 0.104.8 to 0.105.0. I upgraded from Emacs 24.3.1 (the packaged version in Ubuntu 14.04) to a fresh build of Emacs 24.5.1 from source yesterday and did the fresh install of Spacemacs I described in my first comment in this issue.

Emacs *24.5.1* does not work for me on Spacemacs 0.105.1 or 0.105.0 but it works on Spacemacs 0.104.8. Using `--insecure` does not seem to make any difference. I confirmed I can `curl https://melpa.org` from the command line and that works fine.

**Comment by syl20bnr** [comment]
So do we have an issue with GFW ? I need a proxy there to be able to test because sadly my mother in law doesn't know how to start Emacs yet :-)

**Comment by tommyjiang** [comment]
I think we do have an issue with  GFW. 

I will come back to this in the morning (Beijing Time) with some discussion of other Spacemacers in China. 

Maybe I am able to provide a proxy for testing the issue this weekend cause I will be at home then.

**Comment by tommyjiang** [comment]
We have the mirror of `melpa` inside GFW here https://github.com/aborn/popkit-elpa

So we can replace it with the default `melpa`:

    (add-to-list 'package-archives
            '("popkit" . "http://elpa.popkit.org/packages/"))

With this configuration the issue is fixed.

**Comment by BohrShaw** [comment]
So it seems the port 80 is used in tag-v0.104.8 and it can be reached. Why it's changed to 443?

**Comment by scloudyy** [comment]
@tommyjiang I think maybe you can push a pr to Chinese README about the "popkit" then other people will know.

**Comment by robbyoconnor** [comment]
@BohrShaw HTTPS is 443.

**Comment by BohrShaw** [comment]
@robbyoconnor I saw a command like `gnutls-cli --insecure --port 443 melpa.org` is used. And I meant to replace the port 443 with 80.

**Comment by tommyjiang** [comment]
@scloudyy Thanks for your advice. But until now I have not tried the PR function of GitHub.. Maybe I could suggest @syl20bnr update the Chinese `Readme` after he tests this.

**Comment by StreakyCobra** [comment]
@tommyjiang It's a nice opportunity for getting started :wink:

**Comment by royseto** [comment]
I got Spacemacs startup working on Spacemacs 0.105.3, Emacs 24.5.1, and Ubuntu 14.04 just now as follows. I needed the `--insecure` flag (even though I can `curl https://melpa.org` from the command line) and it didn't work using upgrade of my existing Spacemacs install. 

Here are the steps that worked for me:

1. Reboot my host machine (a MacBook Pro running VirtualBox) and start a fresh Ubuntu 14.04 Docker container guest to guarantee a clean networking setup.

2. `cd $HOME`

3. `rm -rf .spacemacs .emacs .emacs.d`

4. `git clone --recursive https://github.com/syl20bnr/spacemacs ~/.emacs.d`

5. `emacs --insecure` (then allow packages to install)

6. Exit Emacs after packages install.

7. Put my customized `$HOME/.spacemacs` file back. Run `emacs` to allow additional packages to install.

**Comment by syl20bnr** [comment]
@royseto you can add:

8. Change value of `dotspacemacs-elpa-https` to `nil` so you won't have to provide `--insecure` everytime you want to upgrade.

@tommyjiang Is the @royseto's procedure work for you ?

**Comment by royseto** [comment]
Thank you @syl20bnr. I will try that before my next upgrade.

**Comment by tommyjiang** [comment]
@syl20bnr  I tested @royseto 's procedure with the latest version of Spacemacs in Beijing and it works fine. 

Maybe you could help to add some instruction in `Readme` for solving this issue.

**Comment by syl20bnr** [comment]
@tommyjiang you mean specifically in the Chinese layer README file ?

I already put instructions in the main README.md and in the FAQ (hence the Fixed tags for this issue).
I can add some notes in Chinese layer since China residents are more exposed to this issue.

**Comment by tommyjiang** [comment]
Yeah, I hope we can add some notes in Chinese layer specifically for China residents. I suppose they may come to know this issue has some relationship with GFW since there is no response when trying to connect to `MELPA`.

@StreakyCobra Maybe I could try it next time :)

**Comment by HIKALU-Z** [comment]
@syl20bnr somebody lived in Europe also meet this problem .but  more people who lived in China meet this problem

**Comment by royseto** [comment]
Just to be clear, I encountered this issue in San Francisco on my home network. No corporate firewalls or other network restrictions.

**Comment by rtz12** [comment]
I also have the exact same issue. No firewall here, curl works fine, Emacs version 24.5.1.

**Comment by beezzdev** [comment]
I am also facing the same issue, from Israel, emacs version 24.5.1.
I am trying from my home network we no firewall. I have tried the --insecure option without success.
The error I get is: error "Package `spacemacs-theme-' is unavailable"

**Comment by sergiokopplin** [comment]
Same problem here in Brazil, emacs v 24.5.1.
Neither "--insecure" or "dotspacemacs-elpa-https" solves the problem. I've proxy in my office.
i've tried the develop branch and another release, but same issue.

**Comment by sergiokopplin** [comment]
log for "emacs":

```
Warning (initialization): An error occurred while loading `/Users/kopplin/.emac\
s.d/init.el':

error: Package `spacemacs-theme-' is unavailable

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the `--debug-init' option to view a complete error backtrace.
```

log for "emacs --debug-init":

```
Debugger entered--Lisp error: (error "Package `spacemacs-theme-' is unavailable")
  signal(error ("Package `spacemacs-theme-' is unavailable"))
  error("Package `%s-%s' is unavailable" spacemacs-theme "")
  package-compute-transaction(nil ((spacemacs-theme)))
  package-install(spacemacs-theme)
  (if pkg-elpa-dir (add-to-list (quote load-path) pkg-elpa-dir) (if log (progn (spacemacs-buffer/append (format "(Bootstrap) Installing %s...\n" pkg)) (spacema$
  (let ((pkg-elpa-dir (spacemacs//get-package-directory pkg))) (if pkg-elpa-dir (add-to-list (quote load-path) pkg-elpa-dir) (if log (progn (spacemacs-buffer/a$
  (condition-case nil (require pkg) (error (require (quote cl)) (let ((pkg-elpa-dir (spacemacs//get-package-directory pkg))) (if pkg-elpa-dir (add-to-list (quo$
  (let ((warning-minimum-level :error)) (condition-case nil (require pkg) (error (require (quote cl)) (let ((pkg-elpa-dir (spacemacs//get-package-directory pkg$
  spacemacs/load-or-install-package(spacemacs-theme)
  (let* ((pkg (spacemacs//get-theme-package theme)) (pkg-dir (spacemacs/load-or-install-package pkg))) (if (or (eq (quote moe-light) theme) (eq (quote moe-dark$
  (cond ((assq theme spacemacs-theme-name-to-package) (let* ((pkg (spacemacs//get-theme-package theme)) (pkg-dir (spacemacs/load-or-install-package pkg))) (if $
  (if (memq theme (custom-available-themes)) nil (cond ((assq theme spacemacs-theme-name-to-package) (let* ((pkg (spacemacs//get-theme-package theme)) (pkg-dir$
  spacemacs/load-theme(spacemacs-dark)
  (let ((default-theme (car dotspacemacs-themes))) (spacemacs/load-theme default-theme) (setq configuration-layer--protected-packages (append (delq nil (mapcar$
  spacemacs/init()
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (message (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version $
  eval-buffer(#<buffer  *load*> nil "/Users/kopplin/.emacs.d/init.el" nil t)  ; Reading at buffer position 1073
  load-with-code-conversion("/Users/kopplin/.emacs.d/init.el" "/Users/kopplin/.emacs.d/init.el" t t)
  load("/Users/kopplin/.emacs.d/init" t t)
```

**Comment by syl20bnr** [comment]
Anybody having the issue, I think this is because of the default time-out of 5sec which is not high enough for some slow connection. There is an error message in case of time-out but maybe it is not displayed correctly since the theme is installed at the very start of Emacs.

I'll see what I can do later today.

**Comment by robbyoconnor** [comment]
I've had this issue -- I upped the timeout to 30 seconds and don't get too many issues

**Comment by sergiokopplin** [comment]
It's working now!

Changed files: 

````
#.spacemacs
dotspacemacs-elpa-https nil

#init.el
(setq url-proxy-services
    '(("no_proxy" . "^\\(proxy.company\\|10.*\\)")
      ("http" . "proxy.company:8080")
      ("https" . "proxy.company:8080")))
(setq url-http-proxy-basic-auth-storage
    (list (list "proxy.company:8080"
        (cons "USERHERE"
              (base64-encode-string "USERHERE:PASS")))))
```

**Comment by robbyoconnor** [comment]
I think this is a proxy/firewall issue -- and not something with melpa like the magit issue prior.

**Comment by ChasingLogic** [comment]
@sergiokopplin did you put that at the top of your init.el or did you add it to the if statement?

**Comment by sergiokopplin** [comment]
after (setq gc-cons-threshold 100000000). init.el

![screen shot 2016-01-25 at 6 26 00 pm](https://cloud.githubusercontent.com/assets/2743180/12563363/1f6c88aa-c391-11e5-8e67-4d1d6aad0341.png)

**Comment by introom** [comment]
dunno how it's related, but just for FYI, I append a zero to the original gc-cons-threshold and then problem solved.

**Comment by zjhmale** [comment]
`--insecure` fix this with v24.5.1

**Comment by sergiokopplin** [comment]
Did you open the issue?

**Comment by r03** [comment]
I had the same problem. Starting with --insecure fixed it.
I'm running Emacs 24.5 on Windows. The windows build doesn't has SSL/TLS by default.
Maybe set dotspacemacs-elpa-https to nil on Windows by default?

**Comment by giodamelio** [comment]
I am having this same problem, and neither using `--insecure` or setting `dotspacemacs-elpa-https` fixes it. I am using Emacs 25.4.1 and Spacemacs 0.105.9.

**Comment by alanct-minted** [comment]
Likewise. `dotspacemacs-elpa-https nil` and `--insecure` aren't doing it for me.

**Comment by alanct-minted** [comment]
FWIW, I was able to get unblocked for the moment by going back to d82224173937a10f1a4594a988444d6e51373322.

```bash
$ cd ~/.emacs.d/
$ git checkout d82224173937a10f1a4594a988444d6e51373322
$ emacs
```

Miiiight still need `dotspacemacs-elpa-https nil` (I had this in my `.spacemacs`) or `emacs --insecure`.

Not sure what the big difference is, but I found that commit from the other conversation in #4097.

**Comment by giodamelio** [comment]
@alanct-minted Thanks, that worked great without `--insecure` or ` dotspacemacs-elpa-https nil`

**Comment by alanct-minted** [comment]
Also, for a bit more context, the exact same configuration ran just fine on my OS X 10.10 host machine on latest `master`, whereas I had to check out the older commit in the Ubuntu VM (running in Vagrant).

**Comment by StreakyCobra** [comment]
@alanct-minted Have you tried with both OS X and Ubuntu VM on the same network (i.e. both at home or both at work, not one at home and one at work)?

**Comment by alanct-minted** [comment]
@StreakyCobra Ugh, good question. I'm 90% sure these were both on the same network. The Mac one is a little difficult to reproduce again since that's my Macbook host, but I'll take a stab at the Ubuntu one both at home and at work today as I make my way into the office and see if either works like the Mac install.

**Comment by alanct-minted** [comment]
Actually, as I typed that... the Ubuntu VM was in Vagrant, which has some inherent complexity creating virtual network interfaces and routing the networking through the host machine. Think it could have anything to do with that extra layer? The Vagrantfile I used for testing [is pretty simple](https://gist.github.com/alanct-minted/65127bc5ba32886885a3). You could probably even `vagrant up` that one and give it a shot on your own machine too.

I'll still play with it on different networks today just to be sure.

**Comment by alanct-minted** [comment]
If you're feeling especially ambitious, you can clone my exact configuration into the Vagrant too:

https://github.com/alanctkc/dotfiles

Then:

```bash
$ cd dotfiles
$ ./install-ubuntu.sh
$ ./dotme.sh
```

**Comment by StreakyCobra** [comment]
The problem is not about the complexity, but because some ISP (internet service providers) or firewalls have caused problems before. Not sure it's this in your case.

I don't really have time these days to test this. Maybe somebody else will do :-)

**Comment by alanct-minted** [comment]
Ok, after testing at home and at work, the network seems to make no difference. Everything works smoothly on OS X, and in an Ubuntu Vivid vagrant VM, packages end up unavailable on emacs startup past d82224173937a10f1a4594a988444d6e51373322.

**Comment by StreakyCobra** [comment]
Ok :-( So I don't really know where it can come from…

**Comment by syl20bnr** [comment]
@alanct-minted can you paste the contents of `SPC h d s` for both machine ?

**Comment by alanct-minted** [comment]
@syl20bnr Here are system info message for both machines: https://gist.github.com/alanct-minted/103a072f04c0219f0dc8

Note, those are the *working* revisions, so the linux one just reports Spacemacs 0.105.0 instead of 0.105.10 since the latter (latest master) was the one that barfed for me on melpa. Interesting the linux machine is on emacs 24.4 instead of 24.5, though.

**Comment by bpizzi** [comment]
I was having the same troubles as you guys (fresh install, emacs 24.5, spacemacs 0.105.16, curl https://melpa.org working, --insecure doing nothing good).
It turned out that my dns configuration wasn't quite ok and that was the root cause of a timeout during the initialization against melpa.
I changed my dns primary server for one that was actually fast for me, and no more "unable to contact melpa.com".

**Comment by cpt0r** [comment]
Had the same issue, was fixed by setting `dotspacemacs-elpa-https nil` in `dotspacemacs/init`.

`--insecure` didn't help, and I think there's a bug somewhere as it seems like that flag is being ignored or overriden.

I confirmed my suspicion and checked the value of  `dotspacemacs-elpa-https` after starting emacs with `--insecure` - sure enough it was set to true, even though the documentation explicitly states:
```
   ;; This variable has no effect if Emacs is launched with the parameter
   ;; `--insecure' which forces the value of this variable to nil.
```

Furthermore, in \*Messages\* buffer I saw:
`... --insecure --port 443 elpa.gnu.org`

...which doesn't make sense to me, port 443 and insecure are mutually exclusive.  In any case if `--insecure` flag was obviously detected then the port shouldn't be set to 443.

**Comment by LachyTech** [comment]
As @syl20bnr suggested earlier about it being a timeout issue: increasing the (dotspacemacs-elpa-timeout) variable in core-dotspacemacs.el to 60s worked for me, even without running with the --insecure argument. I'm in Australia.

**Comment by bingxiao** [comment]
@LachyTech It works for me too. Thanks a lot.

**Comment by danShumway** [comment]
Same results as @Cpt0r. My guess is that ``dotspacemacs-elpa-https`` overrides the ``--insecure`` flag.

**Comment by mimoralea** [comment]
Thanks for the patch @LachyTech. That did it for me too.

**Comment by ruefo** [comment]
Has there been any progress with the Issue? I wanted to give spacemacs a first try but this issue is holding me away from it. I'm behind a corparate proxy and in an ubuntu vm. With my default emacs configuration everything works and connection to melpa is established via the url-proxy-services. I'm somehow helpless, as --debug-init returns nothing here and all fixes above do not work for me. Suggestions?

**Comment by costa** [comment]
TL;DR something happened between https://github.com/syl20bnr/spacemacs/commit/d82224173937a10f1a4594a988444d6e51373322 and the current master screwing the package installation, please debug

First, kudos for this project to all involved, I hope it will keep Emacs on the right track.
Then, this is one major issue I've experienced in the several months that I'm with Spacemacs, so to me, it's of the highest priority. Now I wish I was fluent with Lisp and could contribute, but alas.
Now here's what I just experienced:
- start with very modestly configured (almost default) Spacemacs installation which works well
- add `javascript` layer to `.spacemacs`
- restart Emacs
- watch the bootstrap fail miserably with network errors etc.
- restart Emacs hoping it will go away
- watch the bootstrap fail miserably with network errors etc. *again*
- find this issue, read comments, checkout https://github.com/syl20bnr/spacemacs/commit/d82224173937a10f1a4594a988444d6e51373322
- restart Emacs with mixed feelings
- watch it mostly pass the layer installation and startup
- checkout `master` (aka latest version) back
- restart Emacs out of curiosity
- watch it work flawlessly again

**Comment by syl20bnr** [comment]
Closing this issue since we are now in 0.200.0
@costa please open a fresh issue if you still have problem, thank you!

**Comment by costa** [comment]
@syl20bnr Thanks, it's okay now. The upgrade procedure itself probably has some issues — it doesn't refresh the packages upon switching to the new version (clicking on the UP in the status bar), only on the following manual restart — but it succeeds eventually, so this is clearly not a priority.
